spotify-artist-viewer/Models/SAArtistViewController.m

```diff
@@ -26,28 +30,69 @@ @implementation SAArtistViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     header.text = vcArtist.artistName;
+
     
     NSURL *imageURL = [NSURL URLWithString:vcArtist.imgURL];
     [self.artistImg sd_setImageWithURL:imageURL];
+    self.artistImg.layer.cornerRadius = self.artistImg.frame.size.height/2;
+    [self.artistImg.layer setMasksToBounds:YES];
+
+    UIButton *backButton = [[UIBarButtonItem alloc] initWithTitle:@"Back" style:UIBarButtonItemStyleBordered target:self action:@selector(doneButtonPressed:)];
+    self.navigationItem.backBarButtonItem = backButton;
+ 
+    
+    
+   // while(self.vcArtist.bio == nil){
+       // self.setBio;
+        //self.bio.text = self.vcArtist.bio;
+    //}
+    
+    SARequestManager *manager = [SARequestManager sharedManager];
+    [manager getBio:self.vcArtist.artistUri success:^(NSArray *bios){
+        if(bios.count != NULL){
+            self.vcArtist.bio = [bios objectAtIndex:0];
+            dispatch_async(dispatch_get_main_queue(), ^{
+               self.bio.text = self.vcArtist.bio;
+            });
+            
+        
+        }
+        
+    } failure:^(NSError *error) {
+        NSLog(@"block failure");
+    }];
+    self.bio.text = self.vcArtist.bio;
```

Comment:
You also set the self.bio.text in the completion block. As a result, this line should probably get removed.